mobile/src/full/java/org/openhab/habdroid/core/FcmMessageListenerService.kt

```diff
@@ -202,10 +203,24 @@ class FcmMessageListenerService : FirebaseMessagingService() {
         internal const val EXTRA_NOTIFICATION_ID = "notificationId"
 
         private const val CHANNEL_ID_DEFAULT = "default"
-        private const val SUMMARY_NOTIFICATION_ID = 0
+        internal const val SUMMARY_NOTIFICATION_ID = 0
 
         // Notification grouping is only available on N or higher, as mentioned in
         // https://developer.android.com/guide/topics/ui/notifiers/notifications#bundle
         private val HAS_GROUPING_SUPPORT = Build.VERSION.SDK_INT >= Build.VERSION_CODES.N
+
+        fun clearFcmNotifications(context: Context) {
+            val nm = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+            nm.cancel(SUMMARY_NOTIFICATION_ID)
+            if (HAS_GROUPING_SUPPORT) {
+                val activeNotifications = nm.activeNotifications
+                activeNotifications
+                    .filter { it.groupKey.endsWith("gcm") }
+                    .forEach {
+                        nm.cancel(it.id)
+                        FcmRegistrationService.scheduleHideNotification(context, it.id)
```

Comment:
This is unneeded, as the subnotifications are already cleared. See the handling of the incoming FCM message here: https://github.com/openhab/openhab-android/pull/1594/files#diff-120bc65ca3879a5d1faad0e4bb6e4f65R93

Reply:
Exactly which part of it is unneeded ? `scheduleHideNotification()` for each notification or the entire forEach block ? I'm calling nm.cancel(it.id) to clear locally and scheduling the same id to clear remotely. Which part is redundant ?

Reply:
`scheduleHideNotification` for each notification is redundant, since the FCM hideNotification message with notificationId == SUMMARY_NOTIFICATION_ID will take care of it.

Reply:
I kind of had doubt that just clearing the summary notifications would clear out whole group, so no need to iterate manually. Is that what you talking about ?